libwhisper/WhisperPeer.cpp

```diff
@@ -56,6 +59,12 @@ bool WhisperPeer::interpret(unsigned _id, RLP const& _r)
 
 		if (protocolVersion != version())
 			disable("Invalid protocol version.");
+		else
+		{
+			if (_r.itemCount() > 1) // for backwards compatibility
+				m_bloom = (FixedHash<TopicBloomFilterSize>)_r[1];
+			advertizeTopicsOfInterest();
```

Comment:
Periodic advertisement can be relied on here. Recommend a noteChanged() here as well.